File: MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesApplication.java
package com.novoda.demo.movies;

import android.app.Application;

import com.novoda.demo.movies.Dependencies;
import com.novoda.demo.movies.MovieService;

public class MoviesApplication extends Application {

    private MovieService movieService;

    @Override
    public void onCreate() {
        super.onCreate();
        Dependencies dependencies = new Dependencies(getCacheDir());
        movieService = dependencies.provideMovieService();
    }

    public MovieService movieService() {
        return movieService;
    }
}

File: ModelViewIntentSample/app/src/main/java/com/novoda/movies/mvi/MoviesApplication.kt
package com.novoda.movies.mvi

import android.app.Application
import com.novoda.movies.mvi.search.Dependencies
import com.novoda.movies.mvi.search.Endpoints
import com.novoda.movies.mvi.search.NetworkDependencyProvider
import com.novoda.movies.mvi.search.Secrets


class MoviesApplication : Application(), Dependencies {
    private val secrets = Secrets(BuildConfig.API_KEY)

    override val endpoints = Endpoints(BuildConfig.BASE_URL, BuildConfig.IMAGE_BASE_URL)
    override val networkDependencyProvider = NetworkDependencyProvider(endpoints, secrets)
}
